src/tool_operate.c

```diff
@@ -2197,6 +2254,14 @@ static CURLcode single_transfer(struct GlobalConfig *global,
           my_setopt_str(curl, CURLOPT_ECH, config->ech_config);
 #endif
 
+        /* new in 8.9.0 */
+#if defined(IP_TOS) || defined(IPV6_TCLASS)
+        if(config->ip_tos > 0) {
+          my_setopt(curl, CURLOPT_SOCKOPTFUNCTION, sockopt_callback);
+          my_setopt(curl, CURLOPT_SOCKOPTDATA, config);
+        }
```

Comment:
Would it make sense to display a warning or trace message if a >0 `ip_tos` was requested by the user, but the feature isn't supported?

Reply:
Done